ExomeDepth/make_BEDdetail.py

```diff
@@ -0,0 +1,368 @@
+
+        """ Append color code based on DEL (red) or DUP (blue) """
+        if float(median_ratio) >= 1:  # Assumed DUP
+            color = "0,0,255"  # itemRgb
+        else:  # Assumed DEL
+            color = "255,0,0"  # itemRgb
+
+        event_list.append(color)
+
+        """ Append blockCount, blockSizes, blockStarts. Note that these are not used for ntargets as start/stop for exons is unknown"""
+        event_list.append(1)  # blockCount
+        event_list.append(int(stop)-start)  # blockSizes
+        event_list.append(0)  # blockStarts.
+
+        """ Append custom annotation field in BEDdetail format. 2 additional colums are allowed! """
+        """ custom1 =  sample specific (Correlation and #Calls, split for child and parent). Custom2 = event specific (Ratio and BF) """
+        correlation_child = [float(i) for i in event_dic[item]["child"]["correlation"]]
+        correlation_parent = [float(i) for i in event_dic[item]["parent"]["correlation"]]
+        deldupratio_child = [float(i) for i in event_dic[item]["child"]["deldupratio"]]
+        deldupratio_parent = [float(i) for i in event_dic[item]["parent"]["deldupratio"]]
+        totalcalls_child = [float(i) for i in event_dic[item]["child"]["totalcalls"]]
+        totalcalls_parent = [float(i) for i in event_dic[item]["parent"]["totalcalls"]]
+        bf_child = [float(i) for i in event_dic[item]["child"]["bf"]]
+        bf_parent = [float(i) for i in event_dic[item]["parent"]["bf"]]
+        ratio_child = [float(i) for i in event_dic[item]["child"]["ratio"]]
+        ratio_parent = [float(i) for i in event_dic[item]["parent"]["ratio"]]
+        totalcount_child = int(event_dic[item]["child"]["count"])
+        totalcount_parent = int(event_dic[item]["parent"]["count"])
+
+        """ Calculate Child specific stats """
+        average_correlation_child = "n/a"
+        average_deldupratio_child = "n/a"
+        average_totalcalls_child = "n/a"
+        average_bf_child = "n/a"
+        average_ratio_child = "n/a"
+        bfs_child = "n/a"
+        ratios_child = "n/a"
+        stdev_correlation_child = "n/a"
+        stdev_deldupratio_child = "n/a"
+        stdev_totalcalls_child = "n/a"
+        stdev_bf_child = "n/a"
+        stdev_ratio_child = "n/a"
+
+        if totalcount_child >= 1:
+            bfs_child_list = list(map(float, ['%.1f' % elem for elem in bf_child][0:20]))   # Select first 20 BFs. No sorting yet as this would not be a random subset
+            ratios_child_list = list(map(float, ['%.1f' % elem for elem in ratio_child][0:20]))  # Select first 20 (random) Ratios.  No sorting yet as this would not be a random subset
+            bfs_child_list.sort() # Then sort
+            ratios_child_list.sort() # Then sort
+            bfs_child = ' '.join(map(str, [elem for elem in bfs_child_list])) # Then make string to write in output
+            ratios_child = ' '.join(map(str, [elem for elem in ratios_child_list]))  # Then make string to write in output
```

Comment:
```python
bfs_child = ' '.join(map(str, bfs_child_list)) # Then make string to write in output
ratios_child = ' '.join(map(str, ratios_child_list))  # Then make string to write in output
```

`[elem for elem in bfs_child_list]` doet niets. Bijv:
```python
>>> x
[1, 2, 3]
>>> [y for y in x]
[1, 2, 3]
```
Je kan wel doen, maar map is denk ik beter in dit geval omdat str een bestaande standaard functie is.
```python
>>> [str(y) for y in x]
['1', '2', '3']
```